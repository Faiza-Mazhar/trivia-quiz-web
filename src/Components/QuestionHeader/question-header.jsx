import React from "react";
import "./question-header.style.scss";

const QuestionHeader = ({
  category,
  questionNumber,
  totalQuestions,
  difficulty,
}) => {
  const difficultyLevel =
    difficulty && difficulty.charAt(0).toUpperCase() + difficulty.slice(1);
  return (
    <div className="question-header">
      <div className="category">{`${category}`}</div>
      <div className="question-number">{`Question: ${
        questionNumber + 1
      }/${totalQuestions}`}</div>
      <div className="difficulty">{`${difficultyLevel}`}</div>
    </div>
  );
};

export default QuestionHeader;
