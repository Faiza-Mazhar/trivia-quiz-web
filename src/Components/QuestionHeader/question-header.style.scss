.question-header {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 97%;
  font-weight: bold;
  font-size: medium;
  border-radius: 5px;

  .category {
    width: 33%;
    text-align: left;
  }

  .question-number {
    width: 33%;
    text-align: center;
  }

  .difficulty {
    width: 33%;
    text-align: right;
  }
}
