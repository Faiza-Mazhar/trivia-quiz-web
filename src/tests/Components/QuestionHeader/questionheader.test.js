import QuestionHeader from "../../../Components/QuestionHeader/question-header";
import renderer from "react-test-renderer";

test("Question Header, if no props are set", () => {
  const component = renderer.create(<QuestionHeader />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Question Header, with information", () => {
  const component = renderer.create(
    <QuestionHeader
      category={"Category"}
      questionNumber={1}
      totalQuestions={5}
      difficulty={"hard"}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
