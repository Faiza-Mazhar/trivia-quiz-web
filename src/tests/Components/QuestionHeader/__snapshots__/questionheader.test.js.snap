// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Header, if no props are set 1`] = `
<div
  className="question-header"
>
  <div
    className="category"
  >
    undefined
  </div>
  <div
    className="question-number"
  >
    Question: NaN/undefined
  </div>
  <div
    className="difficulty"
  >
    undefined
  </div>
</div>
`;

exports[`Question Header, with information 1`] = `
<div
  className="question-header"
>
  <div
    className="category"
  >
    Category
  </div>
  <div
    className="question-number"
  >
    Question: 2/5
  </div>
  <div
    className="difficulty"
  >
    Hard
  </div>
</div>
`;
